lib/randomInteger.js

```diff
@@ -0,0 +1,51 @@
+/*
+The operation of the randomInteger function within this file derives from elements within
+https://nodejs.org/api/crypto.html#crypto_crypto_randomint_min_max_callback.  The implementation
+is not designed to be compatible with the original node version, but will make use of an underlying
+native implementation if it exists.
+*/
+
+const crypto = require('crypto');
+
+// Largest integer we can read from a buffer.
+const RAND_MAX = Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
```

Comment:
This presupposes [Node >= 5.10.0](https://nodejs.org/api/buffer.html#buffer_static_method_buffer_from_string_encoding) as do the `let` changes above. It's fine by me but can you verify that this is not going to cause disruption for any libraries that consume this?

We should add the minimum version to package.json.

Reply:
I've switched this to a static def (281474976710655) and rolled back let/const changes - these are var now for wider compatibility.  
I've tested importing the lib in v4, v10 & v15 - v4 imports correctly, but tests do not execute due to mocha engine requirements.